core-api/src/main/java/com/optimizely/ab/optimizelyusercontext/OptimizelyUserContext.java

```diff
@@ -53,44 +68,284 @@ public Optimizely getOptimizely() {
+     * @return All decision results mapped by flag keys.
+     */
     public Map<String, OptimizelyDecision> decideAll(@Nonnull String[] keys) {
         return decideAll(keys, new OptimizelyDecideOption[0]);
     }
 
+    /**
+     * Returns a key-map of decision results ({@link OptimizelyDecision}) for all active flag keys.
+     *
+     * @param options An array of options for decision-making.
+     * @return All decision results mapped by flag keys.
+     */
     public Map<String, OptimizelyDecision> decideAll(@Nonnull OptimizelyDecideOption[] options) {
-        String[] allFlagKeys = {};
+        Map<String, OptimizelyDecision> decisionMap = new HashMap<>();
+
+        ProjectConfig projectConfig = optimizely.getProjectConfig();
+        if (projectConfig == null) {
+            logger.error("Optimizely instance is not valid, failing isFeatureEnabled call.");
+            return decisionMap;
+        }
+
+        List<FeatureFlag> allFlags = projectConfig.getFeatureFlags();
+        String[] allFlagKeys = new String[allFlags.size()];
+        for (int i = 0; i < allFlags.size(); i++) allFlagKeys[i] = allFlags.get(i).getKey();
+
         return decideAll(allFlagKeys, options);
     }
 
+    /**
+     * Returns a key-map of decision results ({@link OptimizelyDecision}) for all active flag keys.
+     *
+     * @return A dictionary of all decision results, mapped by flag keys.
+     */
     public Map<String, OptimizelyDecision> decideAll() {
         return decideAll(new OptimizelyDecideOption[0]);
     }
 
+    /**
+     * Track an event.
+     *
+     * @param eventName The event name.
+     * @param eventTags A map of event tag names to event tag values.
+     * @throws UnknownEventTypeException
+     */
     public void trackEvent(@Nonnull String eventName,
                            @Nonnull Map<String, ?> eventTags) throws UnknownEventTypeException {
         optimizely.track(eventName, userId, attributes, eventTags);
     }
 
+    /**
+     * Track an event.
+     *
+     * @param eventName The event name.
+     * @throws UnknownEventTypeException
+     */
     public void trackEvent(@Nonnull String eventName) throws UnknownEventTypeException {
         trackEvent(eventName, Collections.emptyMap());
     }
 
+    // Utils
+
+    private Map<String, Object> copyAttributes() {
+        return new HashMap<>(attributes);
+    }
+
+    private List<OptimizelyDecideOption> getAllOptions(OptimizelyDecideOption[] options) {
+        List<OptimizelyDecideOption> allOptions = new ArrayList(Arrays.asList(optimizely.defaultDecideOptions));
+        allOptions.addAll(Arrays.asList(options));
+        return allOptions;
+    }
+
+    public static String getFlagKeyInvalidMessage(String flagKey) {
+        return String.format(FLAG_KEY_INVALID, flagKey);
+    }
+
+    public static String getVariableValueInvalidMessage(String variableKey) {
+        return String.format(VARIABLE_VALUE_INVALID, variableKey);
+    }
```

Comment:
One-liner "helper" methods then to add more cognitive overhead than they alleviate. Consider in-lining the String.format directly. Alternatively if you wanted to provide some abstraction, then a customer enum class can be used to define the format functionality:

```java
    private enum Messages {
        FLAG_KEY_INVALID("Flag key \"%s\" is not in datafile."),
        VARIABLE_VALUE_INVALID("Variable value for key \"%s\" is invalid or wrong type.");

        private final String format;

        Messages(String format) {
            this.format = format;
        }

        public String format(Object...args) {
            return String.format(format, args);
        }
    }
```
Which can then be consumed as:
```java
Messages.FLAG_KEY_INVALID.format("key");
```
Note that this implementation reads more direct about what is happening. We're formatting a predefined message.